src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets

```diff
@@ -13,20 +13,37 @@ Copyright (c) .NET Foundation. All rights reserved.
   <PropertyGroup>
 
     <!-- Analysis level is a single property that can be used to control both the compiler warning waves
-         and enable .NET analyzers  -->
+         and enable .NET analyzers. Valid values are 'none', 'latest', 'preview', or a version number  -->
+         
+    <!-- EffectiveAnalysisLevel is used to differentiate from user specified strings (such as 'none')
+         and an implied numerical option (such as '4')-->
+    <EffectiveAnalysisLevel Condition="'$(AnalysisLevel)' == 'none'">4.0</EffectiveAnalysisLevel>
+    <EffectiveAnalysisLevel Condition="'$(AnalysisLevel)' == 'latest'">5.0</EffectiveAnalysisLevel>
+    <EffectiveAnalysisLevel Condition="'$(AnalysisLevel)' == 'preview'">6.0</EffectiveAnalysisLevel>
+    
+    <!-- Automatically set AnalysisLevel if not specified -->
     <AnalysisLevel Condition="'$(AnalysisLevel)' == '' And
                               '$(TargetFrameworkIdentifier)' == '.NETCoreApp' And
                               $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '5.0'))">5.0</AnalysisLevel>
+    <!-- Set EffectiveAnalysisLevel to the value of AnalysisLevel if it is a version number -->
+    <EffectiveAnalysisLevel Condition="'$(EffectiveAnalysisLevel)' == '' And 
+                                       '$(AnalysisLevel)' != ''">$(AnalysisLevel)</EffectiveAnalysisLevel>
 
     <!-- EnableNETAnalyzers Allows analyzes to be disabled in bulk via msbuild if the user wants to -->
     <EnableNETAnalyzers Condition="'$(EnableNETAnalyzers)' == '' And 
-                                   '$(AnalysisLevel)' != '' And
-                                    $([MSBuild]::VersionGreaterThanOrEquals($(AnalysisLevel), '5.0'))">true</EnableNETAnalyzers>
+                                   '$(EffectiveAnalysisLevel)' != '' And
+                                    $([MSBuild]::VersionGreaterThanOrEquals($(EffectiveAnalysisLevel), '5.0'))">true</EnableNETAnalyzers>
     <EnableNETAnalyzers Condition="'$(EnableNETAnalyzers)' == ''">false</EnableNETAnalyzers>
     
-    <!-- Compiler warning level, defaulted to 4. We promote it to 5 if the user has set analysis level to 5 or higher -->
-    <WarningLevel Condition="'$(AnalysisLevel)' != '' And
-                              $([MSBuild]::VersionGreaterThanOrEquals($(AnalysisLevel), '5.0'))">5</WarningLevel>
+    <!-- Compiler warning level, defaulted to 4. We promote it to 5 if the user has set analysis level to 5 or higher
+         NOTE: at this time only the C# compiler supports warning waves -->
+    <WarningLevel Condition="'$(Language)' == 'C#' And
```

Comment:
This will override `WarningLevel` if user sets it to a custom value in their project file, say `6.0` or say `0.0` if they are interested in using a higher `WarningLevel` then `AnalysisLevel` OR turning off compiler warning wave while still retaining analyzer warning wave. I think we need to ensure that we do not override user override of this property.

Reply:
4 is the default. So I can check if the `WarningLevel` is not 4 and only change it in that case. I can't think of any other way to check if a user has set this value or not. It is set regardless.

Reply:
My recommendation is a user wants to do this is to set `<AnalysisLevel>none</AnalysisLevel>` and `<WarningLevel>X</WarningLevel>` with `<EnableNETAnalyzers>true</EnableNETAnalyzers>`

Reply:
What if I want `<AnalysisLevel>5</AnalysisLevel>` and <WarningLevel>6</WarningLevel>?

Reply:
So I want the analyzers from .NET 5 and the compiler warnings from some future version? The only thing I can think of is to not set `WarningLevel` iff it is higher than 4. It will never not be set to some value. The scenario we can't support in this case is the user doing `<AnalysisLevel>5</AnalysisLevel>` and `<WarningLevel>4</WarningLevel>`.

Reply:
Yeah, say we release .net6 with new compiler and analyzer warnings in warning wave 6. I would like to get new compiler warnings, but am not yet ready for new analyzer warnings.

Where does the compiler force set `WarningLevel`? I looked at targets file under https://github.com/dotnet/roslyn/tree/master/src/Compilers/Core/MSBuildTask, but could not find anything. The only default values seem to be in the task https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/MSBuildTask/Csc.cs#L141 and then in the command line parser if no switch was specified.

Reply:
ah, I believe you are right. the compiler itself sets the default not the targets. I'll update this